h2o-algos/src/main/java/hex/svm/LLMatrix.java

```diff
@@ -0,0 +1,87 @@
+package hex.svm;
+
+import water.Iced;
+import water.MemoryManager;
+
+class LLMatrix extends Iced<LLMatrix> {
+
+  private final int _dim;
+  private final double[][] _data;
+
+  LLMatrix(int dim) {
+    _dim = dim;
+    _data = new double[dim][];
+    for (int i = 0; i < dim; i++) {
+      _data[i] = new double[dim - i];
+    }
+  }
+
+  private int dim() {
+    return _dim;
+  }
+
+  final double get(int x, int y) {
+    return _data[y][x-y];
+  }
+
+  final void set(int x, int y, double value) { 
+    _data[y][x-y] = value;
+  }
+
+  final void addUnitMat() {
+    for (double[] col : _data)
+      col[0] += 1;
+  }
```

Comment:
Michal:

This adds 1 to the diagonal element?

Reply:
Yes, adds 1 to the diagonal.